tests/unit_tests/constraint/test_pcalg.py

```diff
@@ -0,0 +1,255 @@
+    assert nx.is_isomorphic(cpdag.sub_undirected_graph(), expected_cpdag.sub_undirected_graph())
+
+
+def test_collider():
+    # construct a causal graph that will result in
+    # x -> y <- z
+    G = nx.DiGraph([("x", "y"), ("z", "y")])
+
+    # construct the SCM and the corresponding causal graph
+    oracle = Oracle(G)
+    pc = PC(ci_estimator=oracle)
+    sample = dummy_sample(G)
+    context = Context(data=sample)
+    pc.fit(context)
+    graph = pc.graph_
+
+    assert graph.has_edge("x", "y", graph.directed_edge_name)
+    assert graph.has_edge("z", "y", graph.directed_edge_name)
+
+
+def test_pc_rule1():
+    edges = [
+        ("x", "y"),
+        ("z", "y"),
+    ]
+    G = ADMG(edges)
+    oracle = Oracle(G)
+    pc = PC(ci_estimator=oracle)
+
+    # If C -> A - B, such that the triple is unshielded,
+    # then orient A - B as A -> B
+
+    G = CPDAG()
+    G.add_edge("A", "B", G.undirected_edge_name)
+    G.add_edge("C", "A", G.directed_edge_name)
+    G_copy = G.copy()
+
+    # First test:
+    # C -> A -> B
+    pc._apply_meek_rule1(G, "A", "B")
+    assert G.has_edge("A", "B", G.directed_edge_name)
+    assert not G.has_edge("A", "B", G.undirected_edge_name)
+    assert G.has_edge("C", "A", G.directed_edge_name)
+
+    # Next, test that nothing occurs
+    # if it is a shielded triple
+    G = G_copy.copy()
+    G.add_edge("C", "B")
+    added_arrows = pc._apply_meek_rule1(G, "A", "B")
+    assert not added_arrows
+
+
+def test_pc_rule2():
+    edges = [
+        ("x", "y"),
+        ("z", "y"),
+    ]
+    G = ADMG(edges)
+    oracle = Oracle(G)
+    pc = PC(ci_estimator=oracle)
+
+    # If C -> A -> B, with C - B
+    # then orient C - B as C -> B
+    G = CPDAG()
+    G.add_edge("C", "B", G.undirected_edge_name)
+    G.add_edge("C", "A", G.directed_edge_name)
+    G.add_edge("A", "B", G.directed_edge_name)
+
+    # First test:
+    # C -> B
+    added_arrows = pc._apply_meek_rule2(G, "C", "B")
+    assert G.has_edge("A", "B", G.directed_edge_name)
+    assert G.has_edge("C", "A", G.directed_edge_name)
+    assert G.has_edge("C", "B", G.directed_edge_name)
+    assert added_arrows
+
+
+def test_pc_rule3():
+    G = ADMG()
+    oracle = Oracle(G)
+    pc = PC(ci_estimator=oracle)
+
+    # If C - A, with C - B and C - D
+    # and B -> A <- D, then orient C - A as C -> A.
+    G = CPDAG()
+    G.add_edge("C", "B", G.undirected_edge_name)
+    G.add_edge("C", "D", G.undirected_edge_name)
+    G.add_edge("C", "A", G.undirected_edge_name)
+    G.add_edge("B", "A", G.directed_edge_name)
+    G.add_edge("D", "A", G.directed_edge_name)
+    G_copy = G.copy()
+
+    added_arrows = pc._apply_meek_rule3(G, "C", "A")
+    assert added_arrows
+    assert G.has_edge("C", "A", G.directed_edge_name)
+
+    # if 'B' and 'D' are adjacent, then it will
+    # not be added
+    G = G_copy.copy()
+    G.add_edge("B", "D", G.directed_edge_name)
+    added_arrows = pc._apply_meek_rule3(G, "C", "A")
+    assert not added_arrows
+
```

Comment:
Can we include some tests that show the expected resolution when the Meek rules collide with fixed edges?

Reply:
I'm not following what you mean here. Can you elaborate @robertness ?

Reply:
I'm just wondering how included and excluded edges, for example, will work with these rules.